tests/suite_test.go

```diff
@@ -176,23 +195,83 @@ func (suite *IntegrationTestSuite) SetupSuite() {
 	require.NoError(suite.T(), err, fmt.Sprintf("Error Deploying DSPA:\n%s", testUtil.PrintConditions(ctx, DSPA, DSPANamespace, clientmgr.k8sClient)))
 	loggr.Info("DSPA Deployed.")
 
-	loggr.Info("Setting up Portforwarding service.")
-	options := []*forwarder.Option{
-		{
-			LocalPort:   PortforwardLocalPort,
-			RemotePort:  APIServerPort,
-			ServiceName: fmt.Sprintf("ds-pipeline-%s", DSPA.Name),
-			Namespace:   DSPANamespace,
-		},
+	if endpointType == "service" {
+
+		loggr.Info("Setting up Portforwarding service.")
+		options := []*forwarder.Option{
+			{
+				LocalPort:   PortforwardLocalPort,
+				RemotePort:  APIServerPort,
+				ServiceName: fmt.Sprintf("ds-pipeline-%s", DSPA.Name),
+				Namespace:   DSPANamespace,
+			},
+		}
+
+		forwarderResult, err = forwarder.WithForwarders(ctx, options, kubeconfig)
+		suite.Require().NoError(err)
+		_, err = forwarderResult.Ready()
+		suite.Require().NoError(err)
+
+		APIServerURL = fmt.Sprintf("http://127.0.0.1:%d", PortforwardLocalPort)
+
+		loggr.Info(fmt.Sprintf("Port forwarding service Successfully set up: %s", APIServerURL))
+
+	} else if endpointType == "route" {
+
+		// Define the namespaced name
+		key := types.NamespacedName{
+			Namespace: DSPANamespace,
+			Name:      "ds-pipeline-" + suite.DSPA.Name,
+		}
+
+		route := &routev1.Route{}
+
+		// Retrieve the router
+		count := 1
+		ok := false
+		for count < 10 {
+			err := clientmgr.k8sClient.Get(context.TODO(), key, route)
+			if err == nil {
+				ok = true
+				break
+			}
+			count++
+			time.Sleep(1 * time.Second)
+		}
+		if !ok {
+			log.Fatal("failed to retrieve router")
+		}
+
+		APIServerURL = fmt.Sprintf("https://%s", route.Status.Ingress[0].Host)
+
+		loggr.Info(fmt.Sprintf("The Api Server URL was retrieved from the Route: %s", APIServerURL))
+
+		suite.Clientmgr.httpClient.Transport = &customTransport{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			},
+			Token: cfg.BearerToken,
+		}
```

Comment:
maybe in the future we can use the dsp custom cert to pass here
that would ensure our dsp custom cert is working properly

Reply:
Exactly. There is a JIRA ( already opened ) to address that